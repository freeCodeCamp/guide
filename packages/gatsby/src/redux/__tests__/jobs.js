const { actions, boundActions } = require(`../actions`)
const jobsReducer = require(`../reducers/jobs`)

Date.now = jest.fn(() => 1482363367071)

describe(`Job actions/reducer`, () => {
  it(`allows creating jobs`, () => {
    expect(actions.createJob({ id: `test job` })).toMatchSnapshot()
  })

  it(`allows completing jobs`, () => {
    let state = jobsReducer(undefined, actions.createJob({ id: `test job` }))
    state = jobsReducer(state, actions.endJob({ id: `test job` }))
    expect(state).toMatchSnapshot()
  })

  it(`allows updating jobs`, () => {
    let state = jobsReducer(undefined, actions.createJob({ id: `test job` }))
    state = jobsReducer(state, actions.setJob({ id: `test job`, progress: 40 }))
    expect(state.active[0].progress).toBeDefined()
  })

  it(`Allows you to set other info on the job`, () => {
    let state = jobsReducer(
      undefined,
      actions.createJob({ id: `test job`, word: `yo` })
    )
    expect(state.active[0].word).toBeDefined()
  })

  it(`throws an error if an ID isn't provided`, done => {
    function runReducer() {
      return jobsReducer(undefined, actions.createJob(`test job`))
    }

    expect(runReducer).toThrowErrorMatchingSnapshot()
    done()
  })

  it(`throws an error if endJob is called for a job that's already ended`, done => {
    function runReducer() {
      let state = jobsReducer(undefined, actions.createJob({ id: `test job` }))
      state = jobsReducer(state, actions.endJob({ id: `test job` }))
      state = jobsReducer(
        state,
        actions.endJob({ id: `test job` }, { name: `test-plugin` })
      )
    }

    expect(runReducer).toThrowErrorMatchingSnapshot()
    done()
  })
})
