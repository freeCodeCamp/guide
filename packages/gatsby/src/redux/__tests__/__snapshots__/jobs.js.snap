// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job actions/reducer allows completing jobs 1`] = `
Object {
  "active": Array [],
  "done": Array [
    Object {
      "completedAt": 1482363367071,
      "createdAt": 1482363367071,
      "id": "test job",
      "plugin": Object {},
      "runTime": 0,
    },
  ],
}
`;

exports[`Job actions/reducer allows creating jobs 1`] = `
Object {
  "payload": Object {
    "id": "test job",
  },
  "plugin": Object {},
  "type": "CREATE_JOB",
}
`;

exports[`Job actions/reducer throws an error if an ID isn't provided 1`] = `"An ID must be provided when creating or setting job"`;

exports[`Job actions/reducer throws an error if endJob is called for a job that's already ended 1`] = `"The plugin \\"test-plugin\\" tried to end a job with the id \\"test job\\" that either hasn't yet been created or has already been ended"`;
